Completion

Smarterr opened the pull request:
Completed the project

Smarterr committed:
Completion

Completed the project

Mitchell560 commented on CMP1903_A1_2324/Testing.cs:
```diff
@@ -10,15 +10,37 @@ namespace CMP1903_A1_2324
+
+        }
+    }
+}
```
Mitchell Review:
Student ID 27607763
Have the requirements been met: There are the 3 required classes being die class, a die class and a testing class. 
The die class holds a value and returns a integer.
3 die are created they are rolled and the are added together to get the sum
Dice rolls cant be done continuously but that is an extra task
The testing class uses the debug.assert() to check if a single die can be bigger than 6 or smaller than 1 which a dice in real life cant do also check if the sum of all dice cant be smaller than 3 and also cant be bigger than 18 
This mean that task 1 is done

Is the code formatted using the Style Guidelines correctly?
The formats follow the way the forked code suggest so that i would say that it is formatted correctly

Is the code easy to read?
The code uses comments to describe how it works and also the variables are named sensibly so I think that it is easily read  

 Are different errors handled correctly?
Yes I mentioned this early when talking about debug.assert()
#debug.assert() to check if a single die can be bigger than 6 or smaller than 1 which a dice in real life cant do also check if the sum of all dice cant be smaller than 3 and also cant be bigger than 18#
since the user can input anything no user error can happen

Mitchell560 reviewed (commented):
Mitchell Review:
Student ID 27607763
Have the requirements been met: There are the 3 required classes being die class, a die class and a testing class.
The die class holds a value and returns a integer.
3 die are created they are rolled and the are added together to get the sum
Dice rolls cant be done continuously but that is an extra task
The testing class uses the debug.assert() to check if a single die can be bigger than 6 or smaller than 1 which a dice in real life cant do also check if the sum of all dice cant be smaller than 3 and also cant be bigger than 18
This mean that task 1 is done

Is the code formatted using the Style Guidelines correctly?
The formats follow the way the forked code suggest so that i would say that it is formatted correctly

Is the code easy to read?
The code uses comments to describe how it works and also the variables are named sensibly so I think that it is easily read

Are different errors handled correctly?
Yes I mentioned this early when talking about debug.assert()
#debug.assert() to check if a single die can be bigger than 6 or smaller than 1 which a dice in real life cant do also check if the sum of all dice cant be smaller than 3 and also cant be bigger than 18#
since the user can input anything no user error can happen

dnkaddw reviewed (commented):
Millie's Review: 
student ID: 27548105

Have the requirements been met:
The requirements for each class is met as said in the brief. 
There are 3 separate classes.
The Game class contain 3 die objects and 3 different values are returned when 'Rolled'.
Debug.Assert() function is used well to verify values are within the correct range 

Is the code formatted using the style guidelines correctly:
I think that this code does follow the style guidelines well.

Is the code easy to read:
The code itself is easy to understand and easy to follow.
Comments also help verify the structure of the code.

Are different errors handled correctly:
As said before, any potential errors are handles well using Debug.Assert() in the Testing class

MatthewPage1 reviewed (commented):
Matthew Review:
Student ID 27608054

Have the requirements been met?: the requirements were to 'prepare a simple application with three classes'. Furthermore, in the die class the requirements are to contain one value of the die and return this to the user. Comparing this to Connor's solution, the requirements have been met as he returns the randomly generated die value back to the screen every time.
Within the game class, Connor creates three dice objects. rolls the three dice and sums and reports the total of the rolls back to the user which shows the requirements have been met. The extra point for this section hasn't been met however, this was not compulsory.
Within the testing class, Connor has made correct use of debug.assert as required within the project requirements aswell as creating a game object.

Is the code formatted using the Style Guidelines Correctly?: I agree that the code is formatted correctly using the Style Guidelines.

Is the code easy to read?: The code is fairly easy to read due to the comments and way the code is laid out. I would only suggest when creating the objects, not to have the same name for the object as the name of the class. For example, this was repeated twice on the testing file for creating the game object 'game game = new Game();' and for creating a die object 'Die die = new Die(new Random());', Other than this minor adjustment, Connor's program is easy to read and understand.

Are different errors handled correctly?: With the use of debug.assert(), Connor checks for errors with his program which are handled as values on a die can only be between 1 and 6 and therefore the sum of 3 die would be between 3 and 18. Therefore I agree that Connor's errors are handled correctly.

Smarterr committed:
Updated Object Names

As suggested in the review from Mathew, I have changed the object names in my testing file
